src/audio/mixer.c

```diff
@@ -140,7 +149,11 @@ static void mix_n_s24(struct comp_dev *dev, struct audio_stream *sink,
 			}
 
 			/* Saturate to 24 bits */
+#if MIXER_HIFI3
+			*dest = AE_SRAI32(AE_SLAI32S(val, 8), 8);
+#else
 			*dest = sat_int24(val);
+#endif
```

Comment:
It would be a lot better if  the ```sat_int24``` implementation had the HIFI implementation.

Reply:
I was thinking that but it would be a quite invasive change. There would be need for saturate from 64b to 24b and 32b to 24b version. 32bi could be with this name, and edit possible uses with 64 bit argument to other components. If you think it's worth the risk (also sat_int16, sat_int32) I could do it. Should I proceed?

Reply:
I'll do this into a separate PR since it impacts nearly every processing component. The saturate optimization was less significant for mixer so it's OK to proceed without.